[Design2-231] Copy Link component to DSCO package

levadadenys opened the pull request:
<!--
  A summary of the change, including any relevant background, motivation, and context.
  If relevant, include a description, screenshots, and/or video of the existing and new behavior.
-->
### [Design2-231] Copy Link component to DSCO package

feat(link): copied link to dsco/link

## Links
- jira ticket: [Design2-231](https://codedotorg.atlassian.net/browse/DESIGN2-231)

<!--
  Links to relevant external resources; ie, specification documents, Jira tickets, related PRs, Honeybadger errors, etc.
-->

<!--
- spec: []()
-->

## Testing story

<!--
  Does your change include appropriate tests?
  If so, please describe how the tests included in this PR are sufficient.
  If not, please explain why this change does not need to be tested.
-->

<!-- Other aspects to consider. Delete any sections that are not relevant to your change. -->

## Deployment strategy

## Follow-up work

<!--
  List (ideally with Jira links) any clean-up or technical debt that will be addressed in future work.
-->

## Privacy

<!--
  1.	Does this change involve the collection, use, or sharing of new Personal Data?
  2.	Does this change involve a new or changed use or sharing of existing Personal Data?
-->

## Security

<!-- Link to Jira task(s) where sensitive security issues are discussed privately. -->

## Caching

## PR Checklist:

<!--
  The final step! Before you create your PR, double-check that everything is in order.
  Change [ ] to [X] during creation to check boxes.
-->

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

kelbyhawn reviewed (commented):
I'm not seeing all of the props that are in the `apps` version of Storybook, is this intended? 
<img width="1039" alt="Screenshot 2025-01-06 at 12 09 24 PM" src="https://github.com/user-attachments/assets/5ecc0f11-292d-408f-abce-6667af4cc2c0" />

Here's what I'm seeing locally:
<img width="1041" alt="Screenshot 2025-01-06 at 12 09 33 PM" src="https://github.com/user-attachments/assets/5ae47ff4-0082-4427-a106-d003e383a6e8" />


Here's what I'm seeing locally:
<img width="1041" alt="Screenshot 2025-01-06 at 12 09 33 PM" src="https://github.com/user-attachments/assets/e5477e91-e1c5-4b77-9917-80ad31d77bba" />

levadadenys commented:
> I'm not seeing all of the props that are in the apps version of Storybook, is this intended?
Thank you for the catch! Yeah, this is a tricky bug that was appearing back and forth, though I thought I fixed it everywhere. Double checked and fixed it here, will check other places you mentioned.

Bug is caused by `import React from 'react'; ` in the component's source code file (because we have global React variable set across the project maybe), and fixed once this import is removed

kelbyhawn reviewed (approved):
Seeing all of the props now! 🎉
